src/com/owncloud/android/lib/resources/files/ChunkedUploadRemoteFileOperation.java

```diff
@@ -161,8 +161,13 @@ protected int uploadFile(OwnCloudClient client) throws IOException {
             }
 
         } finally {
-            SharedPreferences.Editor editor = sharedPref.edit();
-            editor.putStringSet(chunkId, successfulChunks).apply();
+            if (this.isSuccess(status)) {
```

Comment:
wouldn't this part make line 158 obsolete since the finally block takes care of it anyways?